src/shims/foreign_items.rs

```diff
@@ -434,10 +434,19 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
 
             // Target-specific shims
-            _ => match this.tcx.sess.target.target.target_os.as_str() {
-                "linux" | "macos" => return posix::EvalContextExt::emulate_foreign_item_by_name(this, link_name, args, dest, ret),
-                "windows" => return windows::EvalContextExt::emulate_foreign_item_by_name(this, link_name, args, dest, ret),
-                target => throw_unsup_format!("the target `{}` is not supported", target),
+            _ => {
+                match this.tcx.sess.target.target.arch.as_str() {
+                    "x86" | "x86_64" => match link_name {
+                        "llvm.x86.sse2.pause" => return Ok(true),
```

Comment:
I'd rather see an entry `"llvm.x86.sse2.pause"` in the top-level match, and then in there do any further processing that is necessary. As in, I think the match testing here should be inverted.

"Target-specific" was more meant for "target OS"; for now I don't see the need for a target-arch-specific match.

Also, once Miri supports concurrency, should this intrinsic *do* anything? Cc @vakaras

Reply:
The code for this mentions `std::thread::yield_now`, do we need to do anything to also support that? (While we are at it, but doesn't have to be in this PR.)

Reply:
Or alternatively if you want to prepare Miri for target-arch-specific intrinsics, then they should get their own function like the target-OS-specific ones do. (I guess for now that function can live in this file here.)

Reply:
> Also, once Miri supports concurrency, should this intrinsic do anything?

No. At least it does not have to because it is just a recommendation.